### PR TITLE
Load yaml in script

### PR DESCRIPTION
**Pull request by ulich**
This is a preparation to be able to implement https://updaymedia.atlassian.net/browse/DO-142

load yaml spec files in systemtest-env script command and extract information about dns and https instead of passing --alternate-dns and detecting the protocol by looking at elasticbeanstalk resources

- wrap app_config into separate class to be more DRY with default values
- move generation of elasticbeanstalk application name to CloudFormation::Application in order to reuse it

@suxor42 I branched from your script_test_mip branch in order to use what you prepared. I think I found a pretty nice solution organizing the tests, what do you think?

we should not merge https://github.com/as-ideas/yana-baustelle/pull/34/commits anymore, and instead improve on this branch, or we get lots of conflicts :)

### CODE REVIEW

File: lib/baustelle/config/application.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      end
+
+      def https?
+        @raw.fetch('elb', {}).fetch('https', false)
+      end
+
+      def https?
+        @raw.fetch('elb', {}).fetch('https', false)
+      end
+
```
Comment (suxor42) [inline lib/baustelle/config/application.rb]: method https? exists twice.

File: lib/baustelle/config/application.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        @raw.fetch('elb', {})
+      end
+
+      def elb_visibility
+        elb.fetch('visibility', 'external')
+      end
+
+    end
+  end
+end
```
Comment (suxor42) [inline lib/baustelle/config/application.rb]: No newline

File: lib/baustelle/config/application.rb (hunk 1 above)
Comment (ulich) [inline lib/baustelle/config/application.rb]: removed

File: lib/baustelle/config/application.rb (hunk 2 above)
Comment (ulich) [inline lib/baustelle/config/application.rb]: done. Do some ruby versions have issues with no trailing newlines?

Comment (suxor42) [inline lib/baustelle/config/application.rb]: I don't think they have but github marks missing newlines at EOF.

Was just a remark

**Comment by suxor42** [comment]
Look fine now

**Pull request merged**